hapi-server/src/main/java/com/marklogic/util/SearchCriteria.java

```diff
@@ -114,6 +115,32 @@ public static SearchCriteria searchCriteria(String field, TokenOrListParam token
         return new SearchCriteria(field, modifier, values);
     }
 
+    public static List<SearchCriteria> practitionerReferenceAndSearchCriteria(String field, ReferenceAndListParam referenceAndList) {
```

Comment:
It makes sense to put these functions here since it's a collection of static utilities but part of me wonders if these functions should be made more generic here and called by dedicated functions in each resource provider, e.g.:

```java
// PractitionerRoleResourceProvider.java
public class PractitionerRoleResourceProvider implements IResourceProvider {
    // ...
    
    private referenceAndSearchCriteria(String field, ReferenceAndListParam referenceAndList) {
        return SearchCriteria.referenceAndSearchCriteria(field, referenceAndList, "Practitioner");
    }
}
```

```java
// SearchCriteria.java
    public static List<SearchCriteria> referenceAndSearchCriteria(String field, ReferenceAndListParam referenceAndList, String resourceType) {
        List<SearchCriteria> values = new ArrayList<SearchCriteria>();
        if (referenceAndList != null) {
            values = referenceAndList.getValuesAsQueryTokens().stream()
                .map(referenceOrList -> referenceOrSearchCriteria(field, referenceOrList, resourceType))
                .collect(toList());
        }
        return values;
    }
    
    // etc.
```

Some methods would probably need to be renamed for more accuracy, but it avoids needing to refactor later if additional resource providers need to leverage these methods themselves.

Reply:
I agree, I will work on something like this and push the change.